### PR TITLE
Problem indexing

### PR DESCRIPTION
**Pull request by xtalax**

### CODE REVIEW

**Comment by xtalax** [comment]
@ChrisRackauckas this should be finished

File: src/problems/problem_interface.jl [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    else
+        error("Invalid indexing of problem: $sym is not a symbol")
+    end
+end
+
+function Base.setindex!(prob::AbstractSciMLProblem, val, sym)
+    if has_sys(prob.f)
+        if issymbollike(sym)
+            params = getparamsyms(prob)
+            states = Symbol.(setdiff(getsyms(prob), params))
```
Comment (ChrisRackauckas) [inline src/problems/problem_interface.jl:30]: why not just use states?

Comment (xtalax) [reply src/problems/problem_interface.jl:30]: symbolic indexing interface wasn't working, and states is an mtk function

Comment (ChrisRackauckas) [reply src/problems/problem_interface.jl:30]: It's not an MTK function https://github.com/SciML/SymbolicIndexingInterface.jl/blob/master/src/interface.jl#L27

Comment (xtalax) [reply src/problems/problem_interface.jl:30]: if I use states, it can't find the index for some reason

**Review by ChrisRackauckas — APPROVED** [review]

File: src/SciMLBase.jl [hunk 2]
```diff
@@ -10,6 +10,7 @@ using DocStringExtensions
 using LinearAlgebra
 using Statistics
 using Distributed
+using Setfield
```
Comment (ChrisRackauckas) [inline src/SciMLBase.jl:13]:
```suggestion
```

**Pull request merged**